### PR TITLE
Add BlockingFunctionsConfig, RecaptchaConfig and QuotaConfig fields to identityplatform config

### PR DESCRIPTION
**Pull request by mraouffouad**
Add BlockingFunctionsConfig, AuthorizedDomains and QuotaConfig fields to identityplatform config

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
identityplayform: added support for `blocking_functions` `quota` and `authorized_domains` in `google_identity_platform_config`
```

### CODE REVIEW

**Comment by mraouffouad** [comment]
@slevenick This is a redo of [PR/8064](https://github.com/GoogleCloudPlatform/magic-modules/pull/8064) after addressing @roaks3 's comments.

**Review by slevenick — CHANGES_REQUESTED** [review]
googleapi: Error 400: INVALID_CONFIG : SignUp quota must start between now and 365 days from now.

File: mmv1/products/identityplatform/Config.yaml [hunk 1]
```diff
@@ -59,3 +59,70 @@ properties:
...
+            name: 'quota'
+            description: |
+              A sign up APIs quota that customers can override temporarily.
+          - !ruby/object:Api::Type::Time
+            name: 'startTime'
+            description: |
+              When this quota will take affect.
+          - !ruby/object:Api::Type::String
+            name: 'quotaDuration'
+            description: |
```
Comment (rainshen49) [inline mmv1/products/identityplatform/Config.yaml:122]: Should we also add an example of how to structure the duration string? E.g. https://github.com/GoogleCloudPlatform/magic-modules/blob/31547b00e1e93da8fd198d133a17df4264d4eb4d/mmv1/products/alloydb/Cluster.yaml#L171

**Review by slevenick — CHANGES_REQUESTED** [review]
Fails with:
Field 'startTime', Illegal timestamp format; timestamps must end with 'Z' or have a valid timezone offset."

Are you able to run the test yourself?

**Comment by mraouffouad** [comment]
> Fails with: Field 'startTime', Illegal timestamp format; timestamps must end with 'Z' or have a valid timezone offset."
> 
> Are you able to run the test yourself?

I got that error locally before I added the call to UTC(), which I assumed would fix the problem. I'll try a further fix locally and update the PR once it passes.

**Comment by mraouffouad** [comment]
@slevenick Running the test locally passes all the field validation steps but produces the error below at the end. Perhaps it had to do with my env settings. Github is not showing the test on the list failed tests. Can you confirm?

Error:
====
    vcr_utils.go:152: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_identity_platform_config.default will be updated in-place
          ~ resource "google_identity_platform_config" "default" {
                id                         = "projects/terraform-gcip-6/config"
                name                       = "projects/365025219842/config"
                # (3 unchanged attributes hidden)
        
              ~ quota {
                  ~ sign_up_quota_config {
                      ~ start_time     = "2023-07-27T02:55:00Z" -> "2023-07-26T22:55:00-04:00"
                        # (2 unchanged attributes hidden)
                    }
                }
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.

**Comment by slevenick** [comment]
The test is skipped in VCR, so I have to run it myself. I think your local failure indicates a problem with how the resource works. I'm not sure exactly why though.

**Comment by mraouffouad** [comment]
> 

I see. I enabled VCR, let's see what the logs will reveal.

**Comment by mraouffouad** [comment]
> 
Error: Error updating Config "projects/tf-test-my-projectt1rb1914oe/config": Patch "https://identitytoolkit.googleapis.com/v2/projects/tf-test-my-projectt1rb1914oe/config?alt=json&updateMask=autodeleteAnonymousUsers%2CblockingFunctions%2Cquota%2CauthorizedDomains": Requested interaction not found

Ah, I guess this is why VCR was skipped in the first place: https://github.com/hashicorp/terraform-provider-google/issues/14158

It leaves me with no options other than tying to run the test locally. @slevenick I would appreciate it if you do the same.

**Comment by mraouffouad** [comment]
> > 
> 
> Error: Error updating Config "projects/tf-test-my-projectt1rb1914oe/config": Patch "https://identitytoolkit.googleapis.com/v2/projects/tf-test-my-projectt1rb1914oe/config?alt=json&updateMask=autodeleteAnonymousUsers%2CblockingFunctions%2Cquota%2CauthorizedDomains": Requested interaction not found
> 
> Ah, I guess this is why VCR was skipped in the first place: [hashicorp/terraform-provider-google#14158](https://github.com/hashicorp/terraform-provider-google/issues/14158)
> 
> It leaves me with no options other than tying to run the test locally. @slevenick I would appreciate it if you do the same.

However, FWIW the test passed during recording mode.

**Pull request merged**